### PR TITLE
[WIP] Additional settings for serializers.

### PR DESCRIPTION
**Pull request by Scooletz**
Connects to #1694 

Requires at least `NServiceBus  6.0.0-Beta6`.

### CODE REVIEW

**Comment by Scooletz** [comment]
@danielmarbach Is that describes this capability properly or should I move it to the https://github.com/Particular/docs.particular.net/blob/master/nservicebus/serialization/custom-serializer.md and describe it from the custom serializer point of view?

**Comment by danielmarbach** [comment]
I think it would not hurt to show for the custom serializer how to provide such a settings extension method

**Comment by danielmarbach** [comment]
Apart from that I think this addition here is already a good start

**Comment by Scooletz** [comment]
Ping @danielmarbach  I've added settings to the custom serializer. Could you take a look?

File: Snippets/Core/Core_6/Serialization/RegisterCustomSerializer.cs [hunk 1]
```diff
@@ -26,13 +27,19 @@ class MyCustomSerializerDefinition :
...
             }
         }
 
         class MyCustomSerializer :
             IMessageSerializer
         {
+            public MyCustomSerializer(string settingsValue)
+            {
+                // Add code initializing serializer on the basis of settingsValue
+                throw new NotImplementedException();
```
Comment (danielmarbach) [inline Snippets/Core/Core_6/Serialization/RegisterCustomSerializer.cs]: I would remove the throw Exception

**Pull request merged**